deebot_client/commands/xml/map.py

```diff
@@ -0,0 +1,281 @@
+"""Map commands."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from deebot_client.command import Command, CommandResult
+from deebot_client.events import MajorMapEvent, MapSetEvent, MapSetType, MinorMapEvent
+from deebot_client.events.map import CachedMapInfoEvent, MapSubsetEvent
+from deebot_client.logging_filter import get_logger
+from deebot_client.message import HandlingResult, HandlingState
+
+from .common import XmlCommandWithMessageHandling
+
+if TYPE_CHECKING:
+    from typing import Any
+    from xml.etree.ElementTree import Element
+
+    from deebot_client.event_bus import EventBus
+
+_LOGGER = get_logger(__name__)
+
+
+class GetMapSt(XmlCommandWithMessageHandling):
+    """GetMapSt command.
+
+    This command checks whether the current map has been built successfully or not.
+    """
+
+    NAME = "GetMapSt"
+
+    @classmethod
+    def _handle_xml(cls, event_bus: EventBus, xml: Element) -> HandlingResult:
+        """Handle xml message and notify the correct event subscribers.
+
+        Sample message response:
+        "<ctl ret='ok' st='built' method='auto'/>"
+
+        :return: A message response
+        """
+        if xml.attrib.get("ret") != "ok" or not (st := xml.attrib.get("st")):
+            return HandlingResult.analyse()
+
+        built = st == "built"
+        event_bus.notify(CachedMapInfoEvent(name="", active=built))
+        return HandlingResult.success()
+
+
+class GetMapSet(XmlCommandWithMessageHandling):
+    """GetMapSet command.
+
+    This commands gets the list of logical pieces each map is composed of.
+    XML robots do not support multiple maps, so the mid parameter is ignored.
+    """
+
+    _ARGS_MSID = "msid"
+    _ARGS_TYPE = "type"
+    _ARGS_SUBSETS = "subsets"
+
+    NAME = "GetMapSet"
+
+    def __init__(
+        self,
+        # pylint: disable=unused-argument
```

Comment:
Why did you add an unused param?

Reply:
It's a piece of old code, but J've just realized we are missing the code that triggers a GetMapSet from GetMapSt

Reply:
@edenhaus done, I've also made sure GetMapSt calls GetMapSet automatically as it does for the JSON variant